### PR TITLE
Use background task to batch-verify sr25510 signatures with optimum size

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: primitives/io/src/batch_verifier.rs [hunk 1]
```diff
@@ -107,9 +107,30 @@ impl BatchVerifier {
 	) -> bool {
 		if self.invalid.load(AtomicOrdering::Relaxed) { return false; }
 		self.sr25519_items.push(Sr25519BatchItem { signature, pub_key, message });
+
+		if self.sr25519_items.len() >= 128 {
```
Comment (bkchr) [inline primitives/io/src/batch_verifier.rs:111]: Where is this number coming from?

**Comment by burdges** [comment]
Are you sure there is an optimum size?  I'd wishfully imagined https://github.com/dalek-cryptography/curve25519-dalek/pull/249 made arbitrarily large efficient?  Is this CPU cashing issues with large `Vec`s?

File: primitives/io/src/batch_verifier.rs (hunk 1 above)
Comment (NikVolf) [reply primitives/io/src/batch_verifier.rs:111]: I noticed it is linear after 100tx-block

Comment (NikVolf) [reply primitives/io/src/batch_verifier.rs:111]: @burdges ^^

**Pull request merged**